components/accelerator/nvidia/error/xid/component.go

```diff
@@ -203,25 +205,46 @@ func (c *XIDComponent) SetHealthy() error {
 }
 
 func (c *XIDComponent) updateCurrentState() error {
-	osComponent, err := components.GetComponent(os_id.Name)
-	if err != nil {
-		return fmt.Errorf("failed to get os component: %w", err)
-	}
-	osEvents, err := osComponent.Events(c.rootCtx, time.Now().Add(-DefaultRetentionPeriod))
+	rebootEvents, err := getRebootEvents(c.rootCtx)
 	if err != nil {
-		return fmt.Errorf("failed to get os events: %w", err)
+		return fmt.Errorf("failed to get reboot events: %w", err)
 	}
 	localEvents, err := c.store.Get(c.rootCtx, time.Time{})
 	if err != nil {
 		return fmt.Errorf("failed to get all events: %w", err)
 	}
-	events := mergeEvents(osEvents, localEvents)
+	events := mergeEvents(rebootEvents, localEvents)
 	c.mu.Lock()
 	c.currState = EvolveHealthyState(events)
 	c.mu.Unlock()
 	return nil
 }
 
+func getRebootEvents(ctx context.Context) ([]components.Event, error) {
```

Comment:
This seems useful in a lot of places, should we put it in `gpud/tree/main/components/common`?

Reply:
yes we can, did not find a proper file location to put it tho..prob need a `gpud/components/utils`

Reply:
yeah either way is fine :)